### PR TITLE
[#538] add special treatment for module and package info on unloading too

### PR DESCRIPTION
**Pull request by cdietrich**
[#538] add special treatment for module and package info on unloading too

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend [hunk 1]
```diff
@@ -46,7 +46,8 @@ class JavaDerivedStateComputer {
 			unloader.unloadRoot(eObject) 
 		}
 		resource.getContents().clear
-		
+		if (resource.isInfoFile)
```
Comment (szarnekow) [inline org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend]: If I understand this correctly, JavaResources for info files do have a strong reference to a compilation unit.
Should that one be released, too? Assuming we would never have that reference, the null check below would have been sufficient. Other code here could be simplified, too.

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend]: yes and no. i dont see where this reference would be installed in the first place.
or do you mean in other cases?

Comment (szarnekow) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend]: I was thinking about a special treatment a little earlier in the lifecycle, e.g. in `JavaResource.doLoad(InputStream, Map<?, ?>)`

Comment (cdietrich) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend]: i am not sure if that would have any unwanted side effects

Comment (szarnekow) [reply org.eclipse.xtext.java/src/org/eclipse/xtext/java/resource/JavaDerivedStateComputer.xtend]: True. That's also the reason why I approved this minimally invasive fix here.

**Pull request merged**